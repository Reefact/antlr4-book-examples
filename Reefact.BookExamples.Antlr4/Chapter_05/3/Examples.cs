#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using NFluent;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_05._3 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void example_01() {
            // Setup
            GRun grun = GRun.ReadString("a[1]");
            // Exercise
            string lispStyleTree    = grun.ToLispStyleTree();
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(expr a [ (expr 1) ])");
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void example_02() {
            // Setup
            GRun grun = GRun.ReadString("(a[1])");
            // Exercise
            string lispStyleTree    = grun.ToLispStyleTree();
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(expr ( (expr a [ (expr 1) ]) ))");
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void example_03() {
            // Setup
            GRun grun = GRun.ReadString("(((1)))");
            // Exercise
            string output = grun.ToLispStyleTree();
            // Verify
            Check.That(output).IsEqualTo("(expr ( (expr ( (expr ( (expr 1) )) )) ))");
        }

    }

}