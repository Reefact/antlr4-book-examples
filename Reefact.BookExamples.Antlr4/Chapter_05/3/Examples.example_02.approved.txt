graph TD
	1["expr"] --> 2["("]
	1 --> 3["expr"]
	3 --> 4["a"]
	3 --> 5["["]
	3 --> 6["expr"]
	6 --> 7["1"]
	3 --> 8["]"]
	1 --> 9[")"]
